extensions-contrib/kafka-emitter/src/test/java/org/apache/druid/emitter/kafka/EventToJsonSerializerTest.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.emitter.kafka;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.apache.druid.jackson.DefaultObjectMapper;
+import org.apache.druid.java.util.common.DateTimes;
+import org.apache.druid.java.util.common.Intervals;
+import org.apache.druid.java.util.common.granularity.Granularities;
+import org.apache.druid.java.util.emitter.core.Event;
+import org.apache.druid.java.util.emitter.core.EventMap;
+import org.apache.druid.java.util.emitter.service.AlertEvent;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import org.apache.druid.query.TableDataSource;
+import org.apache.druid.query.spec.MultipleIntervalSegmentSpec;
+import org.apache.druid.query.timeseries.TimeseriesQuery;
+import org.apache.druid.segment.VirtualColumns;
+import org.apache.druid.server.QueryStats;
+import org.apache.druid.server.RequestLogLine;
+import org.apache.druid.server.log.DefaultRequestLogEventBuilderFactory;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Collections;
+
+
+public class EventToJsonSerializerTest
```

Comment:
doubt : I couldn't understand why this change is kept in kafka-emitter extension and not core. is there a reason for it?

Reply:
This is a good catch @rohangarg . Previously I have a class that belongs to the kafka-emitter and these were the test for this class. We replace that with EventMap later. One problem with moving the test to the core: DefaultRequestLogEvent is not part of the core, but part of the server.
But I can split the test logic to the corresponding packages and remove this class altogether.

Reply:
Thanks! Yes, the `DefaultRequestLogEvent` test can go in `DefaultRequestLogEventTest` class. It already contains a test for the object serde. probably the tests for other events could also go in their respective test classes.

Reply:
I moved the tests to druid-server (as we need access to the DefaultObjectMapper which is not available in Core) and split it into corresponding classes.